pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -1297,17 +1297,23 @@ func Convert_v1_VirtualMachineInstance_To_api_Domain(vmi *v1.VirtualMachineInsta
 	// in vmi.Spec.Domain.CPU
 	cpuTopology := vcpu.GetCPUTopology(vmi)
 	cpuCount := vcpu.CalculateRequestedVCPUs(cpuTopology)
+	domain.Spec.CPU.Topology = cpuTopology
+	domain.Spec.VCPU = &api.VCPU{
+		Placement: "static",
+		CPUs:      cpuCount,
+	}
 	// set the maximum number of sockets here to allow hot-plug CPUs
 	vmiCPU := vmi.Spec.Domain.CPU
 	if vmiCPU != nil && vmiCPU.MaxSockets != 0 {
 		// Always allow to hotplug to minimum of 1 socket
 		minEnabledCpuCount := cpuTopology.Cores * cpuTopology.Threads
 		// Total vCPU count
 		enabledCpuCount := cpuCount
-		cpuTopology.Sockets = vmiCPU.MaxSockets
-		cpuCount = vcpu.CalculateRequestedVCPUs(cpuTopology)
+		totalCpuTopology := cpuTopology
```

Comment:
nit: `cpuTopology` is a pointer (`*api.CPUTopology`) so actually, changing the `totalCpuTopology`  you are already changing the `domain.Spec.CPU.Topology`

Reply:
Please fix this to avoid future confusion.

Reply:
Seems like more than a nit to me...
If `cpuTopology` and `totalCpuTopology` are two pointers to the same object, what's the point of `totalCpuTopology`?

Reply:
@jean-edouard @fossedihelm Thanks! Will address it

Reply:
@jean-edouard @fossedihelm I used `vcpu.GetCPUTopology(vmi)` to create another object, PTAL :pray: